modules/market/src/main/java/com/opengamma/strata/market/curve/node/CdsCurveNode.java

```diff
@@ -0,0 +1,920 @@
+import com.opengamma.strata.product.credit.ResolvedCdsTrade;
+import com.opengamma.strata.product.credit.type.CdsQuoteConvention;
+import com.opengamma.strata.product.credit.type.CdsTemplate;
+
+/**
+ * A curve node whose instrument is a credit default swap.
+ * <p>
+ * The trade produced by the node will be a protection receiver (BUY) for a positive quantity
+ * and a protection payer (SELL) for a negative quantity.
+ */
+@BeanDefinition
+public final class CdsCurveNode
+    implements CurveNode, ImmutableBean, Serializable {
+
+  /**
+   * The template for the CDS associated with this node.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final CdsTemplate template;
+  /**
+   * The identifier of the market data value that provides the quoted value.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final ObservableId observableId;
+  /**
+   * The label to use for the node.
+   * <p>
+   * When building, this will default based on the tenor if not specified.
+   */
+  @PropertyDefinition(validate = "notEmpty", overrideGet = true)
+  private final String label;
+  /**
+   * The legal entity identifier.
+   * <p>
+   * This identifier is used for the reference legal entity of the CDS.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final StandardId legalEntityId;
+  /**
+   * The node date.
+   * <p>
+   * The node date must be end date.
+   * If this field is null, the end date is computed based on the semi-annual roll convention.
+   */
+  @PropertyDefinition(get = "optional")
+  private final LocalDate endDate;
+  /**
+   * The date order rules.
+   * <p>
+   * This is used to ensure that the dates in the curve are in order.
+   * If not specified, this will default to {@link CurveNodeDateOrder#DEFAULT}.
+   */
+  @PropertyDefinition(validate = "notNull", overrideGet = true)
+  private final CurveNodeDateOrder dateOrder;
+  /**
+   * The market quote convention.
+   * <p>
+   * The CDS is quoted in par spread, points upfront or quoted spread.
+   * See {@link CdsQuoteConvention} for detail.
+   */
+  @PropertyDefinition(validate = "notNull")
+  private final CdsQuoteConvention quoteConvention;
+  /**
+   * The fixed coupon rate.
+   * <p>
+   * This must be represented in decimal form.
+   */
+  @PropertyDefinition(get = "optional")
+  private final Double fixedRate;
+
+  //-------------------------------------------------------------------------
+  /**
+   * Returns a curve node without fixed rate.
+   * <p>
+   * This is typically used for a CDS quoted in par spread or quoted spread.
+   * 
+   * @param template  the template
+   * @param observableId  the observable ID
+   * @param legalEntityId  the legal entity ID
+   * @param quoteConvention  the quote convention
+   * @return the curve node
+   */
+  public static CdsCurveNode of(
+      CdsTemplate template,
+      ObservableId observableId,
+      StandardId legalEntityId,
+      CdsQuoteConvention quoteConvention) {
```

Comment:
Given there are only three conventions, and they have different data, remove these two `of()` methods and add `ofParSpread()`, `ofQuotedSpread()` and `ofPointsUpfront()`. Then there is no need to pass the convention in.